mongodump/mongodump.go

```diff
@@ -48,6 +48,7 @@ type MongoDump struct {
 	authVersion     int
 	archive         *archive.Writer
 	progressManager *progress.Manager
+	count		int
```

Comment:
Unfortunately, making this a struct field for the entire tool won't work. Mongodump dumps collections concurrently, and having a single variable is not thread safe. To scope this more safely to a single goroutine, let's:

1. Make some changes to the progress bar interfaces (see below).
2. Change `func (dump *MongoDump) dumpQueryToWriter` to return a count of documents dumped along with an error.
3. Grab that final count from the progress tracker.